### PR TITLE
Bug 1990489: Reintroduce etcdHighNumberOfFailedGRPCRequests alert for non metal ipi clusters

### PR DESCRIPTION
**Pull request by lilic**
Initially, this alert was reverted as it kept firing on metal ipi clusters in CI. https://bugzilla.redhat.com/show_bug.cgi?id=1990489 For now until we get to the bottom of what is up with etcd on those clusters, and since the alert was not firing on other CI or other types we can reintroduce it back.

/hold
Still need to get a hold of metal ipi cluster to test this, or run the failing CI job.

### CODE REVIEW

**Comment by lilic** [comment]
@smarterclayton can you have a look at the query maybe there is a better way to get the metal ipi type but this is what I came up with. Thanks!

File: manifests/0000_90_etcd-operator_03_prometheusrule.yaml [hunk 1]
```diff
@@ -117,6 +117,31 @@ spec:
         severity: warning
   - name: openshift-etcd.rules
     rules:
+    - alert: etcdHighNumberOfFailedGRPCRequests
+      annotations:
+        description: 'etcd cluster "{{ $labels.job }}": {{ $value }}% of requests for {{ $labels.grpc_method }} failed on etcd instance {{ $labels.instance }}.'
```
Comment (smarterclayton) [inline manifests/0000_90_etcd-operator_03_prometheusrule.yaml:122]: Should this one also have a runbook?

Comment (lilic) [reply manifests/0000_90_etcd-operator_03_prometheusrule.yaml:122]: No it's warning and only critical alerts, for now, have runbooks as per the monitoring team and SRE folks reqs.

**Comment by lilic** [comment]
/test e2e-gcp-five-control-plane-replicas

with test merged this should be green now. We had one failing test due to this alert firing, so retesting...

**Comment by ardaguclu** [comment]
@lilic metal-ipi job is misconfigured. In favor of this PR and future PRs, to assure not breaking blocking jobs, could you please take a look https://github.com/openshift/release/pull/21608?. Thank you.

**Comment by lilic** [comment]
@hexfusion @hasbro17 this is ready to go now. I tested on baremetal installs and it should not fire. 🎉

**Pull request merged**